misc/multiaddr/src/from_url.rs

```diff
@@ -0,0 +1,154 @@
+use crate::{Multiaddr, Protocol};
+use std::{error, fmt, iter, net::IpAddr};
+
+/// Attempts to parse an URL into a multiaddress.
+///
+/// # Example
+///
+/// ```
+/// let addr = parity_multiaddr::from_url("ws://127.0.0.1:8080/").unwrap();
+/// assert_eq!(addr, "/ip4/127.0.0.1/tcp/8080/ws".parse().unwrap());
+/// ```
+///
+pub fn from_url(url: &str) -> std::result::Result<Multiaddr, FromUrlErr> {
+    let url = url::Url::parse(url).map_err(|_| FromUrlErr::BadUrl)?;
+
+    let (protocol, default_port) = match url.scheme() {
+        "ws" => (Protocol::Ws, 80),
+        "wss" => (Protocol::Wss, 443),
+        "http" => (Protocol::Http, 80),
+        "https" => (Protocol::Https, 443),
+        _ => return Err(FromUrlErr::WrongScheme)
```

Comment:
Maybe we should expand the doc comment a bit and mention which schemes are currently supported and also which URL parts are ignored. In fact I think we should error on URLs which contain information we can not preserve in a `Multiaddr`, such as user name, password, query string or fragment. Also I think `UnsupportedScheme` (or just `Unsupported` if we would error on parts we can not handle) would be a better name than `WrongScheme`. Additionally, a function `from_url_lossy` could be offered which ignores the bits we do not care about (essentially what `from_url` does right now).